docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,121 @@
+- Needs to know what the data hosting service endpoint is and how to authenticate into it
+- Needs code changes when an install is attempted to query data
+- Needs to know how to format the data to send to the data hosting service
+- Needs to expose an option for users to opt out of this feature
+
+Additional changes needed:
+- Documentation regarding terms and conditions?
+- Documentation about how to opt out
+- Documentation of requirements: non-disconnected environment
+
+Sending data should not interfere with customer workloads and deployment
+
+### User Stories
+
+#### Story 1
+
+As an infrastructure operator, I can provision and manage my fleet of clusters knowing that data points are automatically collected and sent to the Assisted Installer team without having to do additional configurations. I know my data will be safe and secure and the team will only collect data they need to improve the product. 
+
+
+#### Story 2
+
+As a developer on the assisted installer team, I can analyze the customer data to determine if a feature is worth implementing/keeping/improving. I know that the customer data is accurate and up-to-date. All of the data is parse-able and can be easily tailored to the graphs/visualizations that help my analysis.
+
+#### Story 3
+
+As a product owner, I can determine if the product is moving in the right direction based on the actual customer data. I can help determine which features and bug fixes to prioritize based on the data.
+
+### Implementation Details/Notes/Constraints
+
+Assisted Service changes:
+1. New environment variables to be added:
+    - `DISABLE_DATA_COLLECTION`: If set to `true`, it will disable this feature and thereby allow a customer to opt-out, otherwise the feature is on by default.
+    - `DATA_COLLECTION_URL`: The URL the data will be sent to.
+2. Add additional logic when a cluster reaches one of the following end stages: `error`, `cancelled`, `installed`
+    - Each of these cluster stages represent an end state and that the cluster attempted installation.
+    - Additional logic includes: querying the DB for events of that cluster and streaming this event data to the `DATA_COLLECTION_URL` over HTTPS.
+        - If the customer has large-scale clusters, would it be better to always compress the data in a separate thread?
+3. Add logic for authenticating to the `DATA_COLLECTION_URL` 
+    - Currently unknown since the service to host this data hasn't been decided on yet, but it could take the form of using a Kubernetes Secret
+
+### Risks and Mitigations
+
+On-premises users might have large-scale clusters so network usage and processor usage could be impacted depending on how large the data set is. 
+
+## Design Details
+
+### Open Questions
+
+1. Are we planning to support disconnected environments in the future?
+2. Does any of the event data we store contain Personally Identifiable Information? Does this need to be scrubbed or modified before we push this feature out?
```

Comment:
It's not only the event data, it's also the cluster, infra-env & host data. The only problematic field is the user email and we are hashing it inside the event-scrape so we will need to do the same in the service

Reply:
Good to know! May I have a pointer to how it's hashed right now in event-scrape?

Reply:
https://github.com/openshift-assisted/assisted-events-scrape/blob/master/assisted-events-scrape/utils/anonymizer.py#L25

If we implement it before sending we can then remove it from the scraper (IMO it would be better doing it this way). In such case we'd need to make a plan to roll out without double-encoding or anything like that. WDYT @CrystalChun ?

Reply:
Thanks for the link! Sounds like a good idea to implement the encoding before sending! Good point to make a plan to roll out without double-encoding.